src/packet_analysis/protocol/icmp/ICMPSessionAdapter.cc

```diff
@@ -51,8 +51,13 @@ void ICMPSessionAdapter::UpdateLength(bool is_orig, int len) {
     int& len_stat = is_orig ? request_len : reply_len;
     if ( len_stat < 0 )
         len_stat = len;
-    else
+    else {
         len_stat += len;
+#ifdef DEBUG
+        if ( len_stat < 0 )
+            reporter->Warning("wrapping around for ICMP %s length", is_orig ? "request" : "reply");
+#endif
```

Comment:
Signed integer overflow is undefined behavior in C++ and if we do overflow all bets are off, so this check is too late. If potential overflow is a concern we have to _always check for it before we trigger UB_, e.g.,

```cpp
        // Check for potential overflow.
        //
        // NOTE: Checking `MAX_INT - len` is insufficient since it can
        // underflow for possible negative `len` and/or `len_stat`. Treat
        // potential signs of `len` separately.
        //
        // TODO(bbannier): None of this would be necessary if both `len_stat`
        // and `len` would arrive here unsigned. In that case we wouldn't need
        // to deal with UB since unsigned overflow is defined in C++. This
        // probably would require new validations where these values are
        // extracted.
        if ( len >= 0 ) {
            if ( std::numeric_limits<int>::max() - len > len_stat )
                reporter->FatalError("OH MY!");
        }
        else {
            if ( std::numeric_limits<int>::min() - len > len_stat )
                reporter->FatalError("OH MY!");
        }
        len_stat += len;
```